src/JustSaying/Messaging/Channels/Receive/MessageReceiveController.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace JustSaying.Messaging.Channels.Receive;
+
+public class MessageReceiveController : IMessageReceiveController, IMessageReceiveThreadPausingController
+{
+    private readonly ManualResetEvent _manualResetEvent = new(false);
```

Comment:
This implements `IDisposable` so we should also implement it on this type and tidy up the handle.